File: Posts.cs
using System;
using System.Collections.Generic;
using System.Text;


namespace lab12
{
    class Posts: Workers
    {
        public Posts() { }
        public Posts Set(Name _name, string _post)
        {
            Posts posts = this;
            this.nameP = _name;
            this.postP = _post;
            return posts;
        }
        public void GetPostsFromWorker(Workers worker)
        {
            Posts posts = new Posts();
            Name _name = worker.ReturnName();
            string _post = worker.ReturnPost();
            posts.Set(_name, _post);
        }
        public void Get (Name _name, string _post)
        {
            Posts posts = new Posts();
            posts.Set(_name, _post);
        }
        public void Print ()
        {
            nameP.Print();
            Console.WriteLine($"Должность: {postP}");
        }
        private Name nameP
        {
            set
            {
                _ = nameP;
            }
            get => nameP;
        }
        private string postP
        {
            set
            {
                postP = value;
            }
            get
            {
                return postP;
            }
        }

    }
}

File: Posts.java
package com.company;

import java.util.Scanner;

public class Posts extends Workers
{
    Name name;
    String post;
    public Posts () {};
    public  Posts Set (Name _name, String _post)
    {
        Posts post = this;
        this.name = _name;
        this.post = _post;
        return post;
    }
    public void GetFromWorker (Workers worker)
    {
        Posts post = new Posts();
        Name _name = worker.ReturnName(worker);
        String _post = worker.ReturnPost(worker);
        post.Set(_name, _post);
    }
    public void Get (Name _name)
    {
        Posts post = new Posts();
        String _post;
        Scanner in = new Scanner (System.in);
        System.out.println("Введите должность сотрудника: ");
        _post = in.nextLine();
        post.Set(_name, _post);
    }
    public void Print ()
    {
        System.out.printf("ФИО: %s %s %s\n Должность: %s", name.lastName, name.name, name.patronymic, post);
    }
}
